### ISSUE TITLE
Pass companion file reference to Editor

### ISSUE THREAD

**Issue by adamfowleruk** [adamfowleruk/Paladin#357]
When opening a file in Paladin, pass on the companion file full path to the editor for the file being opened. This assists the Editor when the companion file is not in the same folder.

Only Paladin has an idea of the Project context, and thus file dependencies, of the opened file.

Document this protocol fully and advertise to developers of Editors (E.g. Pe and Koder).

**Comment by scottmc** [comment]
IIRC Pe and perhaps others, also accept the line number to open up to by adding :nnn to the end of the path.

**Comment by adamfowleruk** [comment]
Yeah I'm already passing line number (supported by both Pe and Koder) and column (supported by Koder) - E.g. if you double click on a compilation error.
